apps/server/src/modules/group/uc/group.uc.ts

```diff
@@ -1,304 +1,50 @@
-		const user: UserDO = await this.userService.findById(userId);
-
-		const groups: Page<Group> = await this.groupService.findGroupsByUserAndGroupTypes(user, this.ALLOWED_GROUP_TYPES, {
-			pagination: { skip: 0 },
-		});
-
-		const classInfosFromGroups: ClassInfoDto[] = await this.getClassInfosFromGroups(groups.data);
-
-		return classInfosFromGroups;
-	}
+	public async getGroup(userId: EntityId, groupId: EntityId): Promise<ResolvedGroupDto> {
+		const group: Group = await this.groupService.findById(groupId);
 
-	private async getClassInfosFromGroups(groups: Group[]): Promise<ClassInfoDto[]> {
-		const systemMap: Map<EntityId, SystemDto> = await this.findSystemNamesForGroups(groups);
+		await this.checkPermission(userId, group);
 
-		const classInfosFromGroups: ClassInfoDto[] = await Promise.all(
-			groups.map(async (group: Group): Promise<ClassInfoDto> => this.getClassInfoFromGroup(group, systemMap))
-		);
+		const resolvedUsers: ResolvedGroupUser[] = await this.findUsersForGroup(group);
+		const resolvedGroup: ResolvedGroupDto = GroupUcMapper.mapToResolvedGroupDto(group, resolvedUsers);
 
-		return classInfosFromGroups;
+		return resolvedGroup;
 	}
 
-	private async getClassInfoFromGroup(group: Group, systemMap: Map<EntityId, SystemDto>): Promise<ClassInfoDto> {
-		let system: SystemDto | undefined;
-		if (group.externalSource) {
-			system = systemMap.get(group.externalSource.systemId);
-		}
-
-		const resolvedUsers: ResolvedGroupUser[] = [];
-
-		let synchronizedCourses: Course[] = [];
-		if (this.configService.get('FEATURE_SCHULCONNEX_COURSE_SYNC_ENABLED')) {
-			synchronizedCourses = await this.courseService.findBySyncedGroup(group);
-		}
-
-		const mapped: ClassInfoDto = GroupUcMapper.mapGroupToClassInfoDto(
+	private async checkPermission(userId: EntityId, group: Group): Promise<void> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		return this.authorizationService.checkPermission(
+			user,
 			group,
-			resolvedUsers,
-			synchronizedCourses,
-			system
-		);
-
-		return mapped;
-	}
-
-	private async findSystemNamesForGroups(groups: Group[]): Promise<Map<EntityId, SystemDto>> {
-		const systemIds: EntityId[] = groups
-			.map((group: Group): string | undefined => group.externalSource?.systemId)
-			.filter((systemId: string | undefined): systemId is EntityId => systemId !== undefined);
-
-		const uniqueSystemIds: EntityId[] = Array.from(new Set(systemIds));
-
-		const systems: Map<EntityId, SystemDto> = new Map<EntityId, SystemDto>();
-
-		await Promise.all(
-			uniqueSystemIds.map(async (systemId: string): Promise<void> => {
-				const system: SystemDto = await this.systemService.findById(systemId);
-
-				systems.set(systemId, system);
-			})
+			AuthorizationContextBuilder.read([Permission.GROUP_VIEW])
 		);
-
-		return systems;
 	}
 
 	private async findUsersForGroup(group: Group): Promise<ResolvedGroupUser[]> {
```

Comment:
Now that you split the UC, I see that this method was only used in the getGroup method. I think it is ok when we add the log message back (then we need the logger import). The original was:
```
if (!user) {
	this.logger.warning(new ReferencedEntityNotFoundLoggable(Group.name, group.id, UserDO.name, groupUser.userId));
} else {
	...
}
```